automation/test.sh

```diff
@@ -0,0 +1,268 @@
+while [ $kubectl_rc -ne 0 ] || [ -n "$(_oc get nodes --no-headers | grep NotReady)" ]; do
+    echo "Waiting for all nodes to become ready ..."
+    _oc get nodes --no-headers
+    kubectl_rc=$?
+    sleep 10
+done
+set -e
+
+echo "Nodes are ready:"
+_oc get nodes
+
+_oc describe nodes
+
+_oc adm policy add-scc-to-user privileged system:serviceaccount:kube-system:kubevirt-privileged
+_oc adm policy add-scc-to-user privileged system:serviceaccount:kube-system:kubevirt-controller
+_oc adm policy add-scc-to-user privileged system:serviceaccount:kube-system:kubevirt-apiserver
+
+export VERSION=v0.9.1
+
+_oc create \
+    -f https://github.com/kubevirt/kubevirt/releases/download/$VERSION/kubevirt.yaml
+
+export NAMESPACE="${NAMESPACE:-kube-system}"
+# OpenShift is running important containers under default namespace
+namespaces=(kube-system default)
+if [[ $NAMESPACE != "kube-system" ]]; then
+  namespaces+=($NAMESPACE)
+fi
+
+sample=30
+timeout=300
+
+for i in ${namespaces[@]}; do
+  # Wait until kubevirt pods are running
+  current_time=0
+  while [ -n "$(_oc get pods -n $i --no-headers | grep -v Running)" ]; do
+    echo "Waiting for kubevirt pods to enter the Running state ..."
+    _oc get pods -n $i --no-headers | >&2 grep -v Running || true
+    sleep $sample
+
+    current_time=$((current_time + sample))
+    if [ $current_time -gt $timeout ]; then
+      exit 1
+    fi
+  done
+
+  # Make sure all containers are ready
+  current_time=0
+  while [ -n "$(_oc get pods -n $i -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
+    echo "Waiting for KubeVirt containers to become ready ..."
+    _oc get pods -n $i -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false || true
+    sleep $sample
+
+    current_time=$((current_time + sample))
+    if [ $current_time -gt $timeout ]; then
+      exit 1
+    fi
+  done
+  _oc get pods -n $i
+done
+
+# Prepare PV and PVC for Windows testing, create winrmcli pod
+
+_oc create -f - <<EOF
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: disk-windows
+  labels:
+    kubevirt.io/os: "windows"
+spec:
+  capacity:
+    storage: 30Gi
+  accessModes:
+    - ReadWriteOnce
+  nfs:
+    server: "nfs"
+    path: /
+  storageClassName: local
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: disk-windows
+  labels:
+    kubevirt.io: ""
+spec:
+  storageClassName: local
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 30Gi
+
+  selector:
+    matchLabels:
+      kubevirt.io/os: "windows"
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: winrmcli
+  namespace: default
+spec:
+  containers:
+  - image: kubevirt/winrmcli
+    command:
+      - sleep
+      - "3600"
+    imagePullPolicy: Always
+    name: winrmcli
+restartPolicy: Always
+EOF
+
+# Make sure winrmcli pod is ready
+set +e
+current_time=0
+while ! _oc get pod winrmcli -o yaml | grep "phase: Running" ; do 
```

Comment:
I'm not fond of this kind of checks. Feels fragile. Maybe we can use -o json and jq?

Reply:
updated